src/util/collections.ts

```diff
@@ -24,3 +24,68 @@ export function uniqueMap<K,V>(original: Map<K,V> | null = null, overrides: Map<
   }
   return cloned;
 }
+
+export class BitSet {
+  words!: Uint32Array;
+
+  constructor() {
+    this.clear();
+  }
+
+  get size(): i32 {
+    var count = 0;
+    var words = this.words;
+    for (let i = 0, len = words.length; i < len; i++) {
+      let word = unchecked(words[i]);
+      if (word) count += popcnt(word);
+    }
+    return count;
+  }
+
+  add(index: i32): this {
+    var idx = index >>> 5;
+    var words = this.words;
+    if (idx >= words.length) { // resize
+      this.words = new Uint32Array(idx + 16);
```

Comment:
Should this perhaps double?

Reply:
Should we document, just in case `BitSet` is eventually used for something else as well?

Reply:
Yeah, I'll document usage cases